jmix-flowui/flowui/src/main/java/io/jmix/flowui/component/menusearchfield/MenuSearchField.java

```diff
@@ -0,0 +1,259 @@
+        return new ComponentValueChangeEvent<>(this, this, e.getOldValue(), e.isFromClient());
+    }
+
+    /**
+     * @return true/false if auto select is active/inactive
+     */
+    public boolean isAutoselect() {
+        return getContent().isAutoselect();
+    }
+
+    /**
+     * Set to true to always have the field value automatically
+     * selected when the field gains focus, false otherwise.
+     *
+     * @param autoselect true/false to set auto select on/off
+     */
+    public void setAutoselect(boolean autoselect) {
+        getContent().setAutoselect(autoselect);
+    }
+
+    protected static class MenuItemsTransformer<T extends MenuItem> implements Function<List<T>, List<T>> {
+
+        protected String searchString;
+
+        public String getSearchString() {
+            return searchString;
+        }
+
+        public void setSearchString(String searchString) {
+            this.searchString = searchString;
+        }
+
+        @Override
+        public List<T> apply(List<T> menuItems) {
+            if (Strings.isNullOrEmpty(searchString)) {
+                return menuItems;
+            }
+            Iterator<T> iterator = menuItems.iterator();
+            while (iterator.hasNext()) {
+                T rootItem = iterator.next();
+                boolean match = filterItemRecursive(rootItem, false);
+                if (!match) {
+                    iterator.remove();
+                }
+            }
+            return menuItems;
+        }
+
+        /**
+         * Tests menu item and filters its children recursively by title.
+         *
+         * @param item        item to filter
+         * @param forceRetainAllChildren flag indicating whether to retain all children of this item or not.
+         * @return true/false if the item matches/doesn't match the condition (if title contains search string
+         * or any child matches this condition)
+         */
+        protected boolean filterItemRecursive(MenuItem item, boolean forceRetainAllChildren) {
+            //test item itself
+            boolean selfMatch = testItemMatch(item);
+
+            boolean match;
+            if (item instanceof ParentMenuItem<? extends MenuItem> parentMenuItem) {
+                //test item children and filter them by title
+                //or retain all children if this item matches the condition or force flag is enabled externally
+                boolean anyChildMatch =
+                        filterChildren(parentMenuItem, selfMatch || forceRetainAllChildren);
+
+                match = selfMatch || anyChildMatch;
+                if (match) {
+                    parentMenuItem.setOpened(true);
+                }
+            } else {
+                match = selfMatch;
+            }
+            return match;
+        }
+
+        protected boolean testItemMatch(MenuItem item) {
+            return StringUtils.contains(item.getTitle(), searchString);
```

Comment:
1. Why not `containsIgnoreCase` by defaut? 
2. Some sort of `CASE_INSENSITIVE`/`CASE_SENSITIVE` setting may be helpful.

Reply:
Added search mode for case insensitive search